operator/apis/http/v1alpha1/condition_types.go

```diff
@@ -0,0 +1,88 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:validation:Enum=Ready
+
+// HTTPScaledObjectCreationStatus describes the creation status
+// of the scaler's additional resources such as Services, Ingresses and Deployments
+type HTTPScaledObjectCreationStatus string
+
+const (
+	// Ready indicates the object is fully created
+	Ready HTTPScaledObjectCreationStatus = "Ready"
+)
+
+// +kubebuilder:validation:Enum=ErrorCreatingAppScaledObject;AppScaledObjectCreated;TerminatingResources;AppScaledObjectTerminated;AppScaledObjectTerminationError;PendingCreation;HTTPScaledObjectIsReady;
+
+// HTTPScaledObjectConditionReason describes the reason why the condition transitioned
+type HTTPScaledObjectConditionReason string
+
+const (
+	ErrorCreatingAppScaledObject    HTTPScaledObjectConditionReason = "ErrorCreatingAppScaledObject"
+	AppScaledObjectCreated          HTTPScaledObjectConditionReason = "AppScaledObjectCreated"
+	TerminatingResources            HTTPScaledObjectConditionReason = "TerminatingResources"
+	AppScaledObjectTerminated       HTTPScaledObjectConditionReason = "AppScaledObjectTerminated"
+	AppScaledObjectTerminationError HTTPScaledObjectConditionReason = "AppScaledObjectTerminationError"
+	PendingCreation                 HTTPScaledObjectConditionReason = "PendingCreation"
+	HTTPScaledObjectIsReady         HTTPScaledObjectConditionReason = "HTTPScaledObjectIsReady"
+)
```

Comment:
Same as above, I plan to revisit this in the future to clean these conditions to just the required